src/app/ui/question/question.component.ts

```diff
@@ -46,8 +48,8 @@ export class QuestionComponent implements OnInit {
   }
 
   markAsSolved() {
```

Comment:
Delete this function. It is not necessary at all. When the user clicks the button, open the modal and save the question only after he rates the question or closes the modal. I very much recommend that you will use ng-bootstrap modals like I used in solution component (look there to see an example).

Reply:
I do not agree. Clicking the button should mark this question as solved. Rating is not a must but rather an optional action. What happens if the user clicks the button and then closes his browser? In this case, I still want to save that he had solved the question.

Reply:
Fine, so fix this function too - add NgxSpinner while the writing to the db occurs.

Reply:
Fixed.